### ISSUE TITLE
exportTrailingSlash with hash links produces broken links

### ISSUE THREAD

**Issue by Janpot** [zeit/next.js#9678]
# Bug report

## Describe the bug

exporting pages that contain

```js
<Link href="#hash"><a>text</a></Link>
```
when `exportTrailingSlash: true` is set in `next.config.js` produces links like

```js
<a href="/#hash">text</a>
```

instead of

```js
<a href="#hash">text</a>
```
## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. clone https://github.com/Janpot/hash-link-bug
2. run `yarn && yarn export`
3. verify `href` value in `./out/demo/index.html`
4. See that it contains `/#hash`

## Expected behavior

It should contain `#hash`

## System information

- OS: macOS
- Version of Next.js: 9.1.5

## Additional context

Not sure if this is a valid use of `next/link`, but this works correctly in development mode. It also works correctly when `exportTrailingSlash: false`

**Comment by timneutkens** [comment]
Might not be worth fixing with regards to bundle size. Especially as exportTrailingSlash is sort of a legacy option.

**Comment by lenaggar** [comment]
@timneutkens Is this still open, waiting for a fix?
I'm a first timer, and I'd like to take a look 👀

**Comment by Timer** [comment]
Published as `v9.1.8-canary.7` — please try it out!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
